File: Daily Challanges/2026-04-April/2026-04-25.js
function decompress(str) {
  let mapper = {};
  let decompressed = [];
  const words = str.split(" ");
  for (let i = 0; i < words.length; i++) {
    const word = words[i];
    if (/^\d+$/.test(word)) {
      decompressed.push(mapper[word]);
    } else {
      mapper[`${i + 1}`] = word;
      decompressed.push(word);
    }
  }
  return decompressed.join(" ");
}

console.log(decompress("practice makes perfect and 3 1 2 3"));

File: Daily Challanges/2026-04-April/2026-04-25.py
def decompress(s):
    mapper = {}
    decompressed = []
    for i, word in enumerate(s.split(" ")):
        if word.isdigit():
            decompressed.append(mapper[word])
        else:
            mapper[str(i + 1)] = word
            decompressed.append(word)
    return " ".join(decompressed)


print(decompress("practice makes perfect and 3 1 2 3"))
